tests/test_onnx_v2.py

```diff
@@ -228,28 +196,45 @@ class OnnxExportTestCaseV2(TestCase):
     Integration tests ensuring supported models are correctly exported
     """
 
+    MODELS_TO_TEST = []
+    for (name, model) in PYTORCH_EXPORT_MODELS:
+        for feature, onnx_config_class_constructor in FeaturesManager.get_supported_features_for_model_type(
+            name
+        ).items():
+            MODELS_TO_TEST.append((f"{name}_{feature}", name, model, feature, onnx_config_class_constructor))
+
+    @parameterized.expand(MODELS_TO_TEST)
     @slow
     @require_torch
-    def test_pytorch_export_default(self):
+    def test_pytorch_export(self, test_name, name, model_name, feature, onnx_config_class_constructor):
         from transformers.onnx import export
 
-        for name, model, model_class, config_class, onnx_config_class in PYTORCH_EXPORT_DEFAULT_MODELS:
-            with self.subTest(name):
-                self.assertTrue(hasattr(onnx_config_class, "from_model_config"))
-
-                tokenizer = AutoTokenizer.from_pretrained(model)
-                model = model_class(config_class.from_pretrained(model))
-                onnx_config = onnx_config_class.from_model_config(model.config)
-
-                with NamedTemporaryFile("w") as output:
-                    onnx_inputs, onnx_outputs = export(
-                        tokenizer, model, onnx_config, DEFAULT_ONNX_OPSET, Path(output.name)
-                    )
-
-                    try:
-                        validate_model_outputs(onnx_config, tokenizer, model, Path(output.name), onnx_outputs, 1e-5)
-                    except ValueError as ve:
-                        self.fail(f"{name} -> {ve}")
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        config = AutoConfig.from_pretrained(model_name)
+
+        # Useful for causal lm models that do not use pad tokens.
+        if not getattr(config, "pad_token_id", None):
+            config.pad_token_id = tokenizer.eos_token_id
+
+        model_class = FeaturesManager.get_model_class_for_feature(feature)
+        model = model_class.from_config(config)
```

Comment:
As discussed offline, would it make more sense to loading the model using a pretrained checkpoint instead of the random initialization from a config? 

I think using pretrained weights would be a more realistic test of how the ONNX export is used in real applications. 

But maybe we can leave this as a TODO for a follow-up PR since this one is getting pretty large :)